File: backend/ApiDDD.Domain/Models/Categoria.cs
using ApiDDD.Domain.Models.Enums;
using System.Collections.Generic;

namespace ApiDDD.Domain.Models
{
    /// <summary>
    /// Representa uma categoria para classificar transações financeiras.
    /// A finalidade define quais tipos de transação podem usar esta categoria.
    /// </summary>
    public class Categoria : BaseEntity
    {
        /// <summary>Descrição da categoria. Máximo de 400 caracteres.</summary>
        public string Descricao { get; set; }

        /// <summary>
        /// Define a finalidade da categoria:
        /// DESPESA - somente transações de despesa.
        /// RECEITA - somente transações de receita.
        /// AMBAS - qualquer tipo de transação.
        /// </summary>
        public FinalidadeCategoria Finalidade { get; set; }

        /// <summary>Transações classificadas nesta categoria.</summary>
        public virtual ICollection<Transacao> Transacoes { get; set; } = new List<Transacao>();
    }
}

File: frontend/controle-gastos-front/src/types/categoria.ts
export enum FinalidadeCategoria {
  DESPESA = 1,
  RECEITA = 2,
  AMBAS = 3,
}

export const FinalidadeLabel: Record<FinalidadeCategoria, string> = {
  [FinalidadeCategoria.DESPESA]: "Despesa",
  [FinalidadeCategoria.RECEITA]: "Receita",
  [FinalidadeCategoria.AMBAS]: "Ambas",
};

export interface Categoria {
  id: number;
  descricao: string;
  finalidade: FinalidadeCategoria;
}

export interface CreateCategoriaDTO {
  descricao: string;
  finalidade: FinalidadeCategoria;
}
